### PR TITLE
Read Crucible disks from a JSON file

### PR DESCRIPTION
**Pull request by jmpesp**
Change the cli so that it reads Crucible disks from a JSON file instead
of trying to construct them out of command line arguments.

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -55,9 +58,9 @@ enum Command {
         #[structopt(short, default_value = "1024")]
         memory: u64,
 
-        // Specify multiple disks in groups of 3 targets
-        #[structopt(long, use_delimiter = true)]
-        crucible_target: Vec<SocketAddr>,
+        // file with a JSON array of DiskRequest structs
+        #[structopt(long, parse(from_os_str))]
+        crucible_disks: PathBuf,
```
Comment (pfmooney) [inline cli/src/main.rs:63]: Does this mean that having a crucible disk present is now required?

Comment (jmpesp) [reply cli/src/main.rs]: This is my mistake, made it optional in ec97ee3.

File: cli/src/main.rs [hunk 2]
```diff
@@ -144,28 +147,15 @@ async fn new_instance(
...
-            name: format!("d{}", disk_i),
-            address: target_group.to_vec(),
-            slot: Slot(disk_i),
-            read_only: false,
-            key: None,
-            gen: 0,
-        });
-        disk_i += 1;
+    for disk in &disks {
+        assert_eq!(disk.address.len(), 3);
```
Comment (pfmooney) [inline cli/src/main.rs:151]: Should this perhaps be part of the `parse_crucible_disks`, and then the rest of the validation be left to the API we're calling into?

Comment (jmpesp) [reply cli/src/main.rs]: Moved target validation server side in 00009e6

**Comment by jmpesp** [comment]
@pfmooney let me know if there's any further work here :)

**Comment by pfmooney** [comment]
> @pfmooney let me know if there's any further work here :)

ship it

**Pull request merged**